packages/epo-widget-lib/src/widgets/ColorTool/hooks/useFilteredImages.tsx

```diff
@@ -0,0 +1,76 @@
+import { useLayoutEffect, useState } from "react";
+import isEqual from "lodash/isEqual";
+import { ImageShape } from "@rubin-epo/epo-react-lib/Image";
+import usePrevious from "@/hooks/usePrevious";
+import { ImageFilter } from "../ColorTool";
+import { getFilters, updateColor } from "../lib/canvas";
+
+interface useFilteredImagesProps {
+  images: Array<ImageShape>;
+  filters: Array<ImageFilter>;
+}
+
+const useFilteredImages = ({
+  images,
+  filters,
+}: useFilteredImagesProps): [Array<HTMLCanvasElement>, boolean] => {
+  const previousImages = usePrevious(images);
+  const [imagesLoaded, setImagesLoaded] = useState(0);
+  const [imgElements, setImages] = useState<Array<HTMLImageElement>>();
+
+  const anyImageIncomplete = imgElements?.some(({ complete }) => !complete);
+
+  const loading = imagesLoaded !== images.length && anyImageIncomplete;
+
+  useLayoutEffect(() => {
```

Comment:
trying to figure out how to read cached loading forced a rethink of a this whole setup. All of the images are now loaded as `Image` blobs that are put into an array in the current state, so they do not reload whenever props change.